operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfiguration.java

```diff
@@ -114,4 +115,12 @@ default boolean useFinalizer() {
   default ResourceEventFilter<R> getEventFilter() {
     return ResourceEventFilters.passthrough();
   }
+
+  default long reconciliationMaxInterval() {
```

Comment:
I think it will be better to introduce a separate annotation if we split the duration and its unit to make the name simpler and make it more coherent because it doesn't make sense to provide one without the other.

Reply:
I'm a little worried that it would unnecessary complicate things; also there is always a default value. Also both start with same prefix, I don't think this will be confusing.

Reply:
An alternative is to have the time as a string, like `"10h"` or "140m". But we use this separation in other API already.

Reply:
I don't think it would complicate things and it would make things cleaner to group related things in a single part. Otherwise, you could have annotated reconcilers with just `reconciliationMaxIntervalTimeUnit` and that would look really weird. Currently, the only hint that these things are related are their names. It's better to make it explicit, in my opinion. It also avoids polluting the main configuration with things that will rarely be used. Embedding the unit in the String is a recipe for disaster, imo, and I think it's much cleaner to separate the configuration into a separate annotation.

Reply:
So just to see it clear. The reason to separate this into a new annotation because there are two related fields? 
I'm not sure that makes the effort, especially since we everything in one (for now mandatory) annotation what is a good thing. 
We would like to have this feature by default turned on, so what if that annotation is not present? Actually I guess should be turned off? It's easy to miss that it even exists.

Reply:
Agree that it nicer, but also has these side effects..so I lean little bit more to the current way. But if you insist we can change it. Just lets clarify what should be the behavior.

Reply:
Which side effects?

Reply:
As mentioned, wanted to have this behavior turned on by default. So the developers should turn it off if they are absolutely sure that the operator works as intended (maybe not even that case). But if this is a separate annotation, then it needs to be put there explicitly, otherwise it would be quite contra intuitive to have the functionality turned on without the annotation.

Reply:
The annotation is only needed if you want to change the default values. We already do that in lots of other cases.

The only scenario where things are not clear is if you want to turn it off completely. Then again, that's not clear either with the current implementation.

Reply:
Where do we do that? What I mean is, that in spring and other frameworks  a functionality is turned on by adding an annotation, for me it sounds a little smelly that if we have an annotation for that but its not there and still turned on.

I created a constant for that to turn off, or any non negative value would do. It's quite common to do it this way.

Reply:
Maybe we should ask the users tbh. @andreaTP @lburgazzoli what do you think, should be this a separate annotation or not:
https://github.com/java-operator-sdk/java-operator-sdk/issues/848

Reply:
The namespace control is done that way. Pretty much all the annotation fields have default values so we already turn things on without the annotation being present. This would be the exact same thing here: the child annotation would just allow people to change the default value.

Reply:
To be clear, I'm not saying we should have a completely separate annotation, just a composite child annotation for this.

Reply:
> To be clear, I'm not saying we should have a completely separate annotation, just a composite child annotation for this.

Ahhh, ok that is a completely different story! Sure that makes sense! Sorry....

Reply:
I just understood that I hadn't been clear enough with what I was saying with your last commit :)
Sorry! 😓